packages/react-native/Libraries/Text/Text/RCTTextShadowView.mm

```diff
@@ -123,48 +123,40 @@ - (void)uiManagerWillPerformMounting
-                          }];
-
-  if (maximumLineHeight == 0) {
-    // `lineHeight` was not specified, nothing to do.
-    return;
-  }
-
-  __block CGFloat maximumFontLineHeight = 0;
-
-  [attributedText enumerateAttribute:NSFontAttributeName
-                             inRange:NSMakeRange(0, attributedText.length)
-                             options:NSAttributedStringEnumerationLongestEffectiveRangeNotRequired
-                          usingBlock:^(UIFont *font, NSRange range, __unused BOOL *stop) {
-                            if (!font) {
-                              return;
-                            }
-
-                            if (maximumFontLineHeight <= font.lineHeight) {
-                              maximumFontLineHeight = font.lineHeight;
-                            }
-                          }];
-
-  if (maximumLineHeight < maximumFontLineHeight) {
-    return;
-  }
-
-  CGFloat baseLineOffset = maximumLineHeight / 2.0 - maximumFontLineHeight / 2.0;
-
-  [attributedText addAttribute:NSBaselineOffsetAttributeName
-                         value:@(baseLineOffset)
-                         range:NSMakeRange(0, attributedText.length)];
+    // Retrieve paragraph line height value
+    __block CGFloat paragraphLineHeight = 0;
+    [attributedText enumerateAttribute:NSParagraphStyleAttributeName
+                               inRange:NSMakeRange(0, attributedText.length)
+                               options:NSAttributedStringEnumerationLongestEffectiveRangeNotRequired
+                            usingBlock:^(NSParagraphStyle *paragraphStyle, __unused NSRange range, __unused BOOL *stop) {
+                              if (!paragraphStyle) {
+                                return;
+                              }
+
+                              paragraphLineHeight = paragraphStyle.maximumLineHeight;
+                            }];
+
+    // Retrieve font size and font line height values
+    __block CGFloat fontSize = 0;
+    __block CGFloat fontLineHeight = 0;
+    [attributedText enumerateAttribute:NSFontAttributeName
+                               inRange:NSMakeRange(0, attributedText.length)
+                               options:NSAttributedStringEnumerationLongestEffectiveRangeNotRequired
+                            usingBlock:^(UIFont *font, NSRange range, __unused BOOL *stop) {
+                              if (!font) {
+                                return;
+                              }
+                    
+                              fontSize = font.pointSize;
+                              fontLineHeight = font.lineHeight;
+                            }];
+
+    // Calculate baseline offset based on font size and maximum available line height value
+    CGFloat maximumLineHeight = MAX(paragraphLineHeight, fontLineHeight);
+    CGFloat baselineOffset = (fontSize - maximumLineHeight) / 1.5;
```

Comment:
Could you explain why we were using 2.0 before and now we are using 1.5?

Reply:
Honestly, its quite a "magic" number that can give big enough space to cater text with accent 🤞 
This might be impacting normal text to have bigger white space than before, depends on the font's `pointSize` and `lineHeight` comparison